pyrenew_hew/pyrenew_hew_model.py

```diff
@@ -227,61 +234,89 @@ def sample(
             concentration_rv=self.ed_neg_bin_concentration_rv,
         )
 
-        observed_ed_visits = ed_visit_obs_rv(
+        sampled_ed_visits = ed_visit_obs_rv(
             mu=latent_ed_visits_now,
-            obs=data_observed_disease_ed_visits,
+            obs=data_observed,
         )
 
-        return observed_ed_visits
+        return sampled_ed_visits, iedr
 
 
 class HospAdmitObservationProcess(RandomVariable):
     def __init__(
         self,
         inf_to_hosp_admit_rv: RandomVariable,
         hosp_admit_neg_bin_concentration_rv: RandomVariable,
-    ):
+        ihr_rv: RandomVariable = None,
+        ihr_rel_iedr_rv: RandomVariable = None,
+    ) -> None:
         self.inf_to_hosp_admit_rv = inf_to_hosp_admit_rv
         self.hosp_admit_neg_bin_concentration_rv = (
             hosp_admit_neg_bin_concentration_rv
         )
+        self.ihr_rv = ihr_rv
+        self.ihr_rel_iedr_rv = ihr_rel_iedr_rv
 
     def validate(self):
         pass
 
     def sample(
         self,
         latent_infections: ArrayLike,
+        first_latent_infection_dow: int,
         population_size: int,
-        n_observed_hospital_admissions_datapoints: int,
-        data_observed_disease_hospital_admissions: ArrayLike | None = None,
+        n_datapoints: int,
+        data_observed: ArrayLike = None,
+        iedr: ArrayLike = None,
     ) -> ArrayLike:
         """
         Observe and/or predict incident hospital admissions.
         """
         inf_to_hosp_admit = self.inf_to_hosp_admit_rv()
 
+        if iedr is not None:
+            ihr_rel_iedr = self.ihr_rel_iedr_rv()
+            ihr = iedr[0] * ihr_rel_iedr
+            numpyro.deterministic("ihr", ihr)
+        else:
+            ihr = self.ihr_rv()
+
+        latent_admissions = population_size * ihr * latent_infections
         latent_hospital_admissions = compute_delay_ascertained_incidence(
             p_observed_given_incident=1,
-            latent_incidence=latent_infections,
+            latent_incidence=latent_admissions,
             delay_incidence_to_observation_pmf=inf_to_hosp_admit,
-        )[-n_observed_hospital_admissions_datapoints:]
+        )
+
+        longest_possible_delay = inf_to_hosp_admit.shape[0]
+
+        # we should add functionality to automate this,
+        # along with tests
+        first_latent_admission_dow = (
+            first_latent_infection_dow + longest_possible_delay
+        ) % 7
+
+        predicted_weekly_admissions = daily_to_weekly(
+            latent_hospital_admissions,
+            input_data_first_dow=first_latent_admission_dow,
+            week_start_dow=6,  # MMWR epiweek, starts Sunday
+        )
 
         hospital_admissions_obs_rv = NegativeBinomialObservation(
             "observed_hospital_admissions",
             concentration_rv=self.hosp_admit_neg_bin_concentration_rv,
         )
-        predicted_admissions = latent_hospital_admissions
+
         sampled_admissions = hospital_admissions_obs_rv(
```

Comment:
```suggestion
        observed_hospital_admissions = hospital_admissions_obs_rv(
```
and rename in the return